fre_python_tools/generate_time_averages/generate_time_averages.py

```diff
@@ -0,0 +1,200 @@
+            print('error')
+            exitstatus=1
+        else:
+            print('success')
+            exitstatus=0
+
+    return exitstatus
+
+
+def generate_cdo_timavg(infile=None, outfile=None, avg_type=None):
+    ''' use cdo's python module for time-averaging '''
+    if __debug__:
+        print(f'calling generate_cdo_timavg for file: {infile}')
+        print(f'outfile={outfile}')
+        print(f'avg_type={avg_type}')
+
+    if all([avg_type!='all',avg_type!='seas',avg_type!='month']):
+        print(f'ERROR, avg_type requested unknown.')
+        return 1
+
+    from cdo import Cdo
+    _cdo=Cdo()
+    
+    if avg_type == 'all':
+        print(f'time-averaging requested.')
+        _cdo.timmean(input=infile, output=outfile, returnCdf=True)
+        print(f'time averaging requested')
+    elif avg_type == 'seas':
+        print(f'seasonal time-averaging requested.')
+        _cdo.yseasmean(input=infile, output=outfile, returnCdf=True)
+        print(f'done averaging over seasons')
+    elif avg_type == 'month':
+        print(f'time-averaging requested.')
+        _cdo.ymonmean(input=infile, output=outfile, returnCdf=True)
+        print(f'done averaging over months')
+    else:
+        print(f'problem.')
+        return 1
+
+    print(f'done averaging')
+
+    return 0
+
+
+def generate_time_average(pkg=None, infile=None, outfile=None, avg_type=None):
+    ''' steering function to various averaging functions above'''
+    if __debug__:
+        print(f'calling generate time averages for file: {infile}')
+    exitstatus=1
+
+    #needs a case statement
+    if   pkg == 'cdo'            :
+        exitstatus=generate_cdo_timavg(            infile=infile, outfile=outfile, avg_type=avg_type )
+    elif pkg == 'fre-nctools'    :
+        exitstatus=generate_frenctools_timavg(     infile=infile, outfile=outfile )
+    elif pkg == 'fre-python-tools':
+        exitstatus=generate_frepythontools_timavg( infile=infile, outfile=outfile )
+    else                         :
+        print('requested package unknown. exit.')
+        exitstatus=1
+
+    return exitstatus
+
+def main():
+    ''' main, for steering, when called like `python generate_time_averages.py` '''
+    argparser = argparse.ArgumentParser(
+        description='generate time averages for specified set of netCDF files. Example: \
+        generate-time-averages.py /path/to/your/files/')
+    argparser.add_argument('inf',
+                           help='input file name',
+                           type=str)
+    argparser.add_argument('outf',
+                           help='output file name',
+                           type=str)
+    argparser.add_argument('-p','--pkg',
+                          help='package to use for timavg [e.g. cdo, fre-nctools, fre-python-tools]',
+                          type=str, default='fre-python-tools')
+    argparser.add_argument('-a','--avg',
+                           help='type of time average to generate [e.g. month,seas,all]',
```

Comment:
fre-nctools and native (fre-python-tools) modes don't support the seasonal and monthly modes. Where should that be mentioned?